File: binding/src/charts/libqhbarmodelmapper.cpp
#include <QAbstractBarSeries>
#include <QAbstractItemModel>
#include <QBarModelMapper>
#include <QHBarModelMapper>
#include <QObject>
#include <QString>
#include <QByteArray>
#include <cstring>
#include <qhbarmodelmapper.h>
#include "libqhbarmodelmapper.h"
#include "libqhbarmodelmapper.hxx"

QHBarModelMapper* QHBarModelMapper_new() {
	 return new QHBarModelMapper();
}

QHBarModelMapper* QHBarModelMapper_new2(QObject* parent) {
	 return new QHBarModelMapper(parent);
}

libqt_string QHBarModelMapper_Tr(const char* s) {
	QString _ret = QHBarModelMapper::tr(s);
	QByteArray _b = _ret.toUtf8();
	libqt_string _str;
	_str.len = _b.length();
	_str.data = static_cast<const char*>(malloc(_str.len + 1));
	memcpy((void*)_str.data, _b.data(), _str.len);
	((char*)_str.data)[_str.len] = '\0';
	return _str;
}

QAbstractItemModel* QHBarModelMapper_Model(const QHBarModelMapper* self) {
	return self->model();
}

void QHBarModelMapper_SetModel(QHBarModelMapper* self, QAbstractItemModel* model) {
	self->setModel(model);
}

QAbstractBarSeries* QHBarModelMapper_Series(const QHBarModelMapper* self) {
	return self->series();
}

void QHBarModelMapper_SetSeries(QHBarModelMapper* self, QAbstractBarSeries* series) {
	self->setSeries(series);
}

int QHBarModelMapper_FirstBarSetRow(const QHBarModelMapper* self) {
	return self->firstBarSetRow();
}

void QHBarModelMapper_SetFirstBarSetRow(QHBarModelMapper* self, int firstBarSetRow) {
	self->setFirstBarSetRow(firstBarSetRow);
}

int QHBarModelMapper_LastBarSetRow(const QHBarModelMapper* self) {
	return self->lastBarSetRow();
}

void QHBarModelMapper_SetLastBarSetRow(QHBarModelMapper* self, int lastBarSetRow) {
	self->setLastBarSetRow(lastBarSetRow);
}

int QHBarModelMapper_FirstColumn(const QHBarModelMapper* self) {
	return self->firstColumn();
}

void QHBarModelMapper_SetFirstColumn(QHBarModelMapper* self, int firstColumn) {
	self->setFirstColumn(firstColumn);
}

int QHBarModelMapper_ColumnCount(const QHBarModelMapper* self) {
	return self->columnCount();
}

void QHBarModelMapper_SetColumnCount(QHBarModelMapper* self, int columnCount) {
	self->setColumnCount(columnCount);
}

void QHBarModelMapper_SeriesReplaced(QHBarModelMapper* self) {
	self->seriesReplaced();
}

void QHBarModelMapper_Connect_SeriesReplaced(QHBarModelMapper* self, intptr_t slot) {
    void (*slotFunc)(QHBarModelMapper*) = reinterpret_cast<void (*)(QHBarModelMapper*)>(slot);
    QHBarModelMapper::connect(self, &QHBarModelMapper::seriesReplaced, [self, slotFunc]() {
	slotFunc(self);
    });
}

void QHBarModelMapper_ModelReplaced(QHBarModelMapper* self) {
	self->modelReplaced();
}

void QHBarModelMapper_Connect_ModelReplaced(QHBarModelMapper* self, intptr_t slot) {
    void (*slotFunc)(QHBarModelMapper*) = reinterpret_cast<void (*)(QHBarModelMapper*)>(slot);
    QHBarModelMapper::connect(self, &QHBarModelMapper::modelReplaced, [self, slotFunc]() {
	slotFunc(self);
    });
}

void QHBarModelMapper_FirstBarSetRowChanged(QHBarModelMapper* self) {
	self->firstBarSetRowChanged();
}

void QHBarModelMapper_Connect_FirstBarSetRowChanged(QHBarModelMapper* self, intptr_t slot) {
    void (*slotFunc)(QHBarModelMapper*) = reinterpret_cast<void (*)(QHBarModelMapper*)>(slot);
    QHBarModelMapper::connect(self, &QHBarModelMapper::firstBarSetRowChanged, [self, slotFunc]() {
	slotFunc(self);
    });
}

void QHBarModelMapper_LastBarSetRowChanged(QHBarModelMapper* self) {
	self->lastBarSetRowChanged();
}

void QHBarModelMapper_Connect_LastBarSetRowChanged(QHBarModelMapper* self, intptr_t slot) {
    void (*slotFunc)(QHBarModelMapper*) = reinterpret_cast<void (*)(QHBarModelMapper*)>(slot);
    QHBarModelMapper::connect(self, &QHBarModelMapper::lastBarSetRowChanged, [self, slotFunc]() {
	slotFunc(self);
    });
}

void QHBarModelMapper_FirstColumnChanged(QHBarModelMapper* self) {
	self->firstColumnChanged();
}

void QHBarModelMapper_Connect_FirstColumnChanged(QHBarModelMapper* self, intptr_t slot) {
    void (*slotFunc)(QHBarModelMapper*) = reinterpret_cast<void (*)(QHBarModelMapper*)>(slot);
    QHBarModelMapper::connect(self, &QHBarModelMapper::firstColumnChanged, [self, slotFunc]() {
	slotFunc(self);
    });
}

void QHBarModelMapper_ColumnCountChanged(QHBarModelMapper* self) {
	self->columnCountChanged();
}

void QHBarModelMapper_Connect_ColumnCountChanged(QHBarModelMapper* self, intptr_t slot) {
    void (*slotFunc)(QHBarModelMapper*) = reinterpret_cast<void (*)(QHBarModelMapper*)>(slot);
    QHBarModelMapper::connect(self, &QHBarModelMapper::columnCountChanged, [self, slotFunc]() {
	slotFunc(self);
    });
}

libqt_string QHBarModelMapper_Tr2(const char* s, const char* c) {
	QString _ret = QHBarModelMapper::tr(s, c);
	QByteArray _b = _ret.toUtf8();
	libqt_string _str;
	_str.len = _b.length();
	_str.data = static_cast<const char*>(malloc(_str.len + 1));
	memcpy((void*)_str.data, _b.data(), _str.len);
	((char*)_str.data)[_str.len] = '\0';
	return _str;
}

libqt_string QHBarModelMapper_Tr3(const char* s, const char* c, int n) {
	QString _ret = QHBarModelMapper::tr(s, c, n);
	QByteArray _b = _ret.toUtf8();
	libqt_string _str;
	_str.len = _b.length();
	_str.data = static_cast<const char*>(malloc(_str.len + 1));
	memcpy((void*)_str.data, _b.data(), _str.len);
	((char*)_str.data)[_str.len] = '\0';
	return _str;
}

void QHBarModelMapper_Delete(QHBarModelMapper* self) {
    delete self;
}

File: binding/src/charts/libqhbarmodelmapper.zig
const QtC = @import("qt6zig");
const qtc = @import("qt6c");%%_IMPORTLIBS_%%
%%_STRUCTDEFS_%%

/// https://doc.qt.io/qt-6/qhbarmodelmapper.html
pub const qhbarmodelmapper = struct {

    /// New constructs a new QHBarModelMapper object.
    pub fn New() QtC.QHBarModelMapper {
        return qtc.QHBarModelMapper_new();
    }


    /// New2 constructs a new QHBarModelMapper object.
    pub fn New2(parent: ?*anyopaque) QtC.QHBarModelMapper {
        return qtc.QHBarModelMapper_new2(@ptrCast(parent));
    }


    pub fn Tr(s: []const u8, allocator: std.mem.Allocator) []const u8 {
        const _str = qtc.QHBarModelMapper_Tr(@ptrCast(s));
defer qtc.libqt_string_free(&_str);
const _ret = allocator.alloc(u8, _str.len) catch @panic("qhbarmodelmapper.Tr: Memory allocation failed");
@memcpy(_ret, _str.data[0.._str.len]);
return  _ret;
    }

    pub fn Model(self: ?*anyopaque, ) QtC.QAbstractItemModel {
        return qtc.QHBarModelMapper_Model(@ptrCast(self));
    }

    pub fn SetModel(self: ?*anyopaque, model: ?*anyopaque) void {
        qtc.QHBarModelMapper_SetModel(@ptrCast(self), @ptrCast(model));
    }

    pub fn Series(self: ?*anyopaque, ) QtC.QAbstractBarSeries {
        return qtc.QHBarModelMapper_Series(@ptrCast(self));
    }

    pub fn SetSeries(self: ?*anyopaque, series: ?*anyopaque) void {
        qtc.QHBarModelMapper_SetSeries(@ptrCast(self), @ptrCast(series));
    }

    pub fn FirstBarSetRow(self: ?*anyopaque, ) i32 {
        return qtc.QHBarModelMapper_FirstBarSetRow(@ptrCast(self));
    }

    pub fn SetFirstBarSetRow(self: ?*anyopaque, firstBarSetRow: i32) void {
        qtc.QHBarModelMapper_SetFirstBarSetRow(@ptrCast(self), firstBarSetRow);
    }

    pub fn LastBarSetRow(self: ?*anyopaque, ) i32 {
        return qtc.QHBarModelMapper_LastBarSetRow(@ptrCast(self));
    }

    pub fn SetLastBarSetRow(self: ?*anyopaque, lastBarSetRow: i32) void {
        qtc.QHBarModelMapper_SetLastBarSetRow(@ptrCast(self), lastBarSetRow);
    }

    pub fn FirstColumn(self: ?*anyopaque, ) i32 {
        return qtc.QHBarModelMapper_FirstColumn(@ptrCast(self));
    }

    pub fn SetFirstColumn(self: ?*anyopaque, firstColumn: i32) void {
        qtc.QHBarModelMapper_SetFirstColumn(@ptrCast(self), firstColumn);
    }

    pub fn ColumnCount(self: ?*anyopaque, ) i32 {
        return qtc.QHBarModelMapper_ColumnCount(@ptrCast(self));
    }

    pub fn SetColumnCount(self: ?*anyopaque, columnCount: i32) void {
        qtc.QHBarModelMapper_SetColumnCount(@ptrCast(self), columnCount);
    }

    pub fn SeriesReplaced(self: ?*anyopaque, ) void {
        qtc.QHBarModelMapper_SeriesReplaced(@ptrCast(self));
    }

    pub fn OnSeriesReplaced(self: ?*anyopaque, callback: *const fn (?*anyopaque) callconv(.c) void) void {
        qtc.QHBarModelMapper_Connect_SeriesReplaced(@ptrCast(self), @intCast(@intFromPtr(callback)));
    }

    pub fn ModelReplaced(self: ?*anyopaque, ) void {
        qtc.QHBarModelMapper_ModelReplaced(@ptrCast(self));
    }

    pub fn OnModelReplaced(self: ?*anyopaque, callback: *const fn (?*anyopaque) callconv(.c) void) void {
        qtc.QHBarModelMapper_Connect_ModelReplaced(@ptrCast(self), @intCast(@intFromPtr(callback)));
    }

    pub fn FirstBarSetRowChanged(self: ?*anyopaque, ) void {
        qtc.QHBarModelMapper_FirstBarSetRowChanged(@ptrCast(self));
    }

    pub fn OnFirstBarSetRowChanged(self: ?*anyopaque, callback: *const fn (?*anyopaque) callconv(.c) void) void {
        qtc.QHBarModelMapper_Connect_FirstBarSetRowChanged(@ptrCast(self), @intCast(@intFromPtr(callback)));
    }

    pub fn LastBarSetRowChanged(self: ?*anyopaque, ) void {
        qtc.QHBarModelMapper_LastBarSetRowChanged(@ptrCast(self));
    }

    pub fn OnLastBarSetRowChanged(self: ?*anyopaque, callback: *const fn (?*anyopaque) callconv(.c) void) void {
        qtc.QHBarModelMapper_Connect_LastBarSetRowChanged(@ptrCast(self), @intCast(@intFromPtr(callback)));
    }

    pub fn FirstColumnChanged(self: ?*anyopaque, ) void {
        qtc.QHBarModelMapper_FirstColumnChanged(@ptrCast(self));
    }

    pub fn OnFirstColumnChanged(self: ?*anyopaque, callback: *const fn (?*anyopaque) callconv(.c) void) void {
        qtc.QHBarModelMapper_Connect_FirstColumnChanged(@ptrCast(self), @intCast(@intFromPtr(callback)));
    }

    pub fn ColumnCountChanged(self: ?*anyopaque, ) void {
        qtc.QHBarModelMapper_ColumnCountChanged(@ptrCast(self));
    }

    pub fn OnColumnCountChanged(self: ?*anyopaque, callback: *const fn (?*anyopaque) callconv(.c) void) void {
        qtc.QHBarModelMapper_Connect_ColumnCountChanged(@ptrCast(self), @intCast(@intFromPtr(callback)));
    }

    pub fn Tr2(s: []const u8, c: []const u8, allocator: std.mem.Allocator) []const u8 {
        const _str = qtc.QHBarModelMapper_Tr2(@ptrCast(s), @ptrCast(c));
defer qtc.libqt_string_free(&_str);
const _ret = allocator.alloc(u8, _str.len) catch @panic("qhbarmodelmapper.Tr2: Memory allocation failed");
@memcpy(_ret, _str.data[0.._str.len]);
return  _ret;
    }

    pub fn Tr3(s: []const u8, c: []const u8, n: i32, allocator: std.mem.Allocator) []const u8 {
        const _str = qtc.QHBarModelMapper_Tr3(@ptrCast(s), @ptrCast(c), n);
defer qtc.libqt_string_free(&_str);
const _ret = allocator.alloc(u8, _str.len) catch @panic("qhbarmodelmapper.Tr3: Memory allocation failed");
@memcpy(_ret, _str.data[0.._str.len]);
return  _ret;
    }

    /// Delete this object from C++ memory.
    pub fn QDelete(self: ?*anyopaque) void {
        qtc.QHBarModelMapper_Delete(@ptrCast(self));
    }
};
